### PR TITLE
Feature/mn 366 upgrade torch

### PR DESCRIPTION
**Pull request by ahirner**
# MN-366 Upgrade Torch

## How to test

On a GPU machine with `cuda >= 11.0`: 

```shell
docker pull moonvision/dev-moonbox:cuda-latest
docker run -it --rm --gpus all moonvision/dev-moonbox:cuda-latest python -c "\
import torch, torchvision;\
assert torch.cuda.is_available();\
print(torch.__version__, torchvision.__version__)
"
```
Expect: `1.11.0+cu113 0.12.0+cu113`.

### CODE REVIEW

File: docker/python-base/install_conda.sh [hunk 1]
```diff
@@ -1,22 +1,24 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
-wget -q https://repo.anaconda.com/miniconda/Miniconda3-py38_4.9.2-Linux-x86_64.sh -O ~/miniconda.sh
-/bin/bash ~/miniconda.sh -b -p /opt/conda
-rm ~/miniconda.sh
-ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh
-/opt/conda/bin/conda update -n base -c defaults conda
-/opt/conda/bin/conda init bash
+curl -L -O https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-$(uname)-$(uname -m).sh
```
Comment (ahirner) [inline docker/python-base/install_conda.sh:4]: [Miniforge](https://github.com/conda-forge/miniforge) makes it harder to mess up conda channel preference.

[mamba](https://github.com/mamba-org/mamba) has crucial tooling for dependency fixes, like `mamba repoquery whoneeds`.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ env:
   DOCKER_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
   # default prefix for all pushed images,
   # set if you need to push images to another set of repos
-  IMAGE_PRE: ''
+  IMAGE_PRE: 'dev-'
```
Comment (truh) [inline .github/workflows/build.yml]: Will you revert this?

File: docker/python-base/environment.yml [hunk 3]
```diff
@@ -3,20 +3,20 @@ name: base
 #
 #  Dependencies include a small set of build-time
 #  dependencies to ease compiling libraries.
+channels:
+  - conda-forge
 dependencies:
-  - blas=1.0
+  - libblas=*=*mkl
```
Comment (truh) [inline docker/python-base/environment.yml:12]: what does  `=*=*` mean? can you add a comment

Comment (ahirner) [reply docker/python-base/environment.yml:12]: done

File: .github/workflows/build.yml (hunk 2 above)
Comment (ahirner) [reply .github/workflows/build.yml]: Yes, before merging downstream PRs. This requires approval of all related PRs.

**Review by truh — APPROVED** [review]

**Comment by truh** [comment]
Looks good in isolation. Might unapprove if I find a problem in the other PRs which might be traced down to changes here.

**Comment by ahirner** [comment]
Thx for the fixes in master. I'll perform the merges/changes across repos outside working hours later this week.

**Pull request merged**